### PR TITLE
ActionList: Make filtering opt in

### PR DESCRIPTION
**Pull request by MaxCloutier**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/workflows-merchant-guidance/issues/660

The auto-showing of filtering added the search bar in unwanted areas, making it opt-in will make sure it doesn't show up unless specified.

### WHAT is this pull request doing?

Adding a prop on the Header and on ActionList to enable filtering when a list has 8 or more items. The reason Header requires the prop is that ActionList is deeply nested under it and it need to be specified there.

### 🎩 
- Add a few items in the secondaryActions or actionGroups in DetailsPage.tsx in the playground to make the total above 8.
- Make sure that the filtering shows up
- Go to the ActionList story
- Add items to be above 8
- Filtering shouldn't be present
- Add the `hasFiltering` prop to ActionList
- Filtering should be present

### CODE REVIEW

File: polaris-react/src/components/ActionList/ActionList.tsx [hunk 1]
```diff
@@ -20,6 +21,8 @@ export interface ActionListProps {
   sections?: readonly ActionListSection[];
   /** Defines a specific role attribute for each action in the list */
   actionRole?: 'menuitem' | string;
+  /** Allow users to filter items in the list. Will only show if more than 8 items in the list. The item content of every items must be a string for this to work */
+  hasFiltering?: boolean;
```
Comment (kyledurand) [inline polaris-react/src/components/ActionList/ActionList.tsx]: What about 
```suggestion
  allowFiltering?: boolean;
```

File: .changeset/lazy-hornets-train.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Make ActionList filtering opt-in
```
Comment (kyledurand) [inline .changeset/lazy-hornets-train.md]:
```suggestion
Added `allowFiltering` prop on `ActionList`, and `filterActions` prop on Page Header
```

File: .changeset/lazy-hornets-train.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
```
Comment (kyledurand) [inline .changeset/lazy-hornets-train.md]: We're adding props here which is always a minor version bump. Patch is reserved for bug fixes, minor visual changes etc...

```suggestion
'@shopify/polaris': minor
```

**Review by kyledurand — COMMENTED** [review]
This is looking good. Can you add the prop to the .stories file that's supposed to show the filter field and also `/snapit` and share the spin link?

**Comment by MaxCloutier** [comment]
> This is looking good. Can you add the prop to the .stories file that's supposed to show the filter field and also `/snapit` and share the spin link?

@kyledurand here you go! https://admin.web.web-xuhr.maxime-cloutier.us.spin.dev/store/shop1

File: polaris-react/src/components/ActionList/ActionList.tsx [hunk 4]
```diff
@@ -132,11 +130,17 @@ export function ActionList({
...
   }, [filteredSections]);
 
-  const showSearch = totalActions >= 8;
+  const totalActions =
+    finalSections?.reduce(
+      (acc: number, section) => acc + section.items.length,
+      0,
+    ) || 0;
+
+  const hasSearch = totalActions >= 8;
```
Comment (kyledurand) [inline polaris-react/src/components/ActionList/ActionList.tsx]: This variable name doesn't make as much sense as it used to. Can we rename it and also move `8` to a const up near the top of the file? Just above the function decl

```suggestion
  const hasManyActions = totalActions >= FILTER_ACTIONS_THRESHOLD;
```

**Review by kyledurand — APPROVED** [review]
Just some minor comments but this looks good! 👍

**Pull request merged**